engine/runtime-integration-tests/src/test/java/org/enso/interpreter/node/expression/builtin/meta/TypeChainTest.java

```diff
@@ -0,0 +1,104 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.test.utils.ContextUtils;
+import org.graalvm.polyglot.Value;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class TypeChainTest {
+  @ClassRule public static final ContextUtils ctx = ContextUtils.createDefault();
+  private static Value typeOf;
+
+  @BeforeClass
+  public static void initTypeOf() {
+    typeOf =
+        ctx.evalModule(
+            """
+            import Standard.Base.Meta
+
+            main = Meta.type_of
+            """);
+  }
+
+  @Test
+  public void textChain() {
+    var type = typeOf.execute("Hello World!");
+    var raw = (Type) ctx.unwrapValue(type);
+    var all = raw.allTypes(ctx.ensoContext());
+
+    var exp1 = ctx.ensoContext().getBuiltins().text();
+    var exp2 = ctx.ensoContext().getBuiltins().any();
+    assertArrayEquals("Text type and Any", new Object[] {exp1, exp2}, all);
+  }
+
+  @Test
+  public void textTypeChain() {
+    var textType = typeOf.execute("Ciao");
+    var textTypeType = typeOf.execute(textType);
+    var raw = (Type) ctx.unwrapValue(textTypeType);
+    var all = raw.allTypes(ctx.ensoContext());
+
+    var exp1 = ctx.ensoContext().getBuiltins().text().getEigentype();
+    var exp2 = ctx.ensoContext().getBuiltins().any();
+    assertArrayEquals("Text.type and Any", new Object[] {exp1, exp2}, all);
```

Comment:
- is this correct - in the [light of this test](https://github.com/enso-org/enso/pull/13961/files#r2401345452)?
- shouldn't rather be `Type.type` and `Any.type`?
- why should a parent type of a value by `Any` and not `Any.type`?
- btw. if this is wrong, then it is _not a heritage_, but my mistake - I wrote the `allTypes` method